### DISCUSSION TITLE
当你的技术栈与软工交集为空（x

### DISCUSSION THREAD

**Discussion by giscus** [ouuan/iles-blog#51] [Comments]
# 当你的技术栈与软工交集为空（x

2023 年春季学期清华大学软件工程课程的游记，记录一下课程的学习过程，在小作业与大作业中遇到的问题。

https://ouuan.moe/post/2023/03/thuse

**Comment by SunnyCloudYang** [comment]
学到了（指details的样式x）

**Comment by Chiyuru** [comment]
> ~~所以说 X 自信就是要允许 Y 才能有，以 X 自信为名打压 Y 到底是什么操作~~

想起来了某个书院带头的纯数应数之争（x

**Comment by YfLi03** [comment]
读完此文想起了这个 https://www.zhihu.com/question/447160870/answer/2903761444  （（

ouuan 大佬何不加入网络部 迅速成为软工助教

**Reply by ouuan** [reply]
软工大作业这种级别的项目我还真没写过（

以后如果要我来当助教，首先至少得允许小作业用 Vue，只能 Vue 或者可以选都行（

**Reply by YfLi03** [reply]
感觉也很好啊！前端小作业 Vue React 二选一肯定不会有人反对，只要有人愿意出题（

**Reply by ouuan** [reply]
> 前端小作业 Vue React 二选一肯定不会有人反对

现在的前端鄙视链不是 React 在最底层，Vue 退出讨论吗，要是有同学 Twitter 混的多说不定就会反对（x

感觉最好是能提供保姆式教学和 fly bitch 两个版本的小作业，可以选择完全不 follow 具体技术栈的要求。只不过我虽然不想写 React，但这个小作业要填空的部分实在是太少了，我可能也懒得从零开始写一个完整的（

**Reply by YfLi03** [reply]
以后或许可以可以用科协暑培作业替代软工小作业？暑培作业自由度应该比较大

**Reply by Ashitemaru** [reply]
首先有个点，就是这学期已经是科协暑培作业代替软工作业的学期了，然而由于暑培几乎完全没有留档作业，所以只能重新命题。然而holder虽然成功向课程组论证了这个方案的可行性（当然也没少被拷打类似万一科协不想出题了咋办之类），但根据实际工作来看，课程组也只是名义上会接受，实际上依然是随便找几个人随便玩玩小作业而已。
这学期的小作业改革工作从上学期末做到这学期都快上第一堂课才开始内测，说实话已经做好挨打准备了，当时已经讨论过各种突然冒出来c2w、gamma、holder1之类和我们当时嫌弃vue作业一样嫌弃这份作业的状况。
如果有dalao能愿意接着这份破玩意往上做，还是推倒再来，感觉课程组不会介意，你只要愿意接这个摊子，那就来吧。

**Reply by YfLi03** [reply]
诶 所以什么叫 “ 这学期已经是科协暑培作业代替软工作业的学期了”

**Reply by Ashitemaru** [reply]
因为我之前带过暑培，而在此之前八九零三届软工使用的vue小作业因为用vue2且文档年久失修已经诟病已久（零字班是问题集中爆发的一届，当时c7w找我吐槽这件事，所以我才有改这玩意的想法），而用暑培作业代替软工小作业的评论也一直有。在我副主席退休的最后一次团会上，时任科协主席xq就提出了科协可以接这个单子，相当于官方认可了。
暑假之后我就以课咨委身份去和软工课程组谈条件去了（x 最后课程组接受了这一方案（不如说是因为软工小作业他们本身就不当回事，谁来弄都一样），所以理论上这一学期就已经是这一方案的实践学期，如果有需要的话我理论上可以提供当时的策划案等各种。
不过这个方案他是否有效是要看长期的，这也是我现在觉得最不靠谱的，因为课程组也没有任何强宣称一两年之后你科协要是想换小作业他们就一定让换。而且这事本身就没有充分宣传出去，估计大家还以为这次就是课程组自娱自乐把holder拉过来命题的。
说难听点，我不到半年都毕业了，我死后管他洪水滔天，但要真说不如ouuan来当副主席把这个东西接下去，反正副主席想来就能来（x

**Reply by Ashitemaru** [reply]
要我说的话，软工现在真的最缺的就是这种非大作业助教，大作业助教很多那种顺水淌不负责任空吃每月2k6的。你只要和课程组说你有心做这些事，他们给你八抬大轿抬进去（

**Reply by YfLi03** [reply]
诶，学长（们）辛苦了

**Comment by c7w** [comment]
骂 holder 骂的比我多 我赢了（逃

**Reply by Ashitemaru** [reply]
还真是

**Comment by Ashitemaru** [comment]
holder来挨打了。
首先承认自己确实忘clear了，网络也确实只给了最简化最简化的。
其次是ESLint和码风这些，emm，holder确实prefer了一个大众之外的码风（x
总之也确实...这确实是个欠打磨的小作业，有的地方确乎给的太简陋，所以dalao这学期上完有无成为助教的打算

**Reply by ouuan** [reply]
到时候再说吧（

起码可以来修一下小作业，可能加个 Vue 的版本；大作业助教不好说（
